From f16834b763eb7052e477e0671c91254151f81a69 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Wed, 27 Dec 2017 06:12:50 +0000
Subject: [PATCH] wr-policy: allow syslogd_t search cgroup_t dirs

Reference sources: selinux-policy-3.13.1-225.6.fc25.src.rpm
Changes come from: policy-f25-base.patch

Allow systemd-journal to search /sys/fs/cgroup/* to fix:
  [FAILED] Failed to start Journal Service.

Fixed avc denials:
  avc: denied { search } for pid=123 comm="systemd-journal" \
  name="/" dev="tmpfs" ino=9126 \
  scontext=system_u:system_r:syslogd_t:s15:c0.c1023 \
  tcontext=system_u:object_r:cgroup_t:s0 tclass=dir

  avc: denied { getattr } for pid=123 comm="systemd-journal" \
  name="user@0.service" dev="cgroup2" ino=350 \
  scontext=system_u:system_r:syslogd_t:s15:c0.c1023 \
  tcontext=system_u:object_r:cgroup_t:s0 tclass=dir

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/logging.te | 1 +
 1 file changed, 1 insertion(+)

--- refpolicy.orig/policy/modules/system/logging.te
+++ refpolicy/policy/modules/system/logging.te
@@ -501,6 +501,7 @@ files_var_lib_filetrans(syslogd_t, syslo
 fs_getattr_all_fs(syslogd_t)
 fs_search_auto_mountpoints(syslogd_t)
 fs_search_tmpfs(syslogd_t)
+fs_search_cgroup_dirs(syslogd_t)
 
 mls_file_read_all_levels(syslogd_t)
 mls_file_write_all_levels(syslogd_t) # Need to be able to write to /var/run/ and /var/log directories
